andela/magma-backend
# #167749957 One-way trip endpoint

#### What does this PR do?

This PR implements the one-way trip request endpoint

#### Description of Tasks

- Setup test suite for this endpoint
- Add a schema for validating the request body
- Optimize `getErrorLabels` function
- Create a service class to mediate between database and request controller
- Create a controller class to implement one-way trip request endpoint
- Validate request body
- Add the route
- Test with postman
- Refactor tests to ensure all cases are passing
- Document endpoint using Swagger


#### How should this be manually tested?

- Clone the Repo, change into the directory and run `npm install`
- Run `npm test` to see all test cases
- Run `npm dev` to start the server
- Make a `POST` request to `http://localhost:{PORT}/api/v1/requests` on Postman using valid credentials

Sample request body
```
{
    "origin": "Lagos",
    "destination": "London",
    "type": "one-way",
    "departureDate": "2020-10-11",
    "reason": "Moving to a new office location",
    "accommodation": "Buckingham Palace"
}
```

#### Relevant pivotal tracker story

[167749957](https://www.pivotaltracker.com/story/show/167749957)



#### Screenshots

![Screenshot (143)](https://user-images.githubusercontent.com/43535158/64486010-7fb68d00-d21f-11e9-8a53-b3edd989151f.png)
